storage/client/src/samples/java/blob/BasicExample.java

```diff
@@ -0,0 +1,111 @@
+import com.azure.storage.blob.StorageClient;
+import com.azure.storage.blob.models.ListBlobsOptions;
+import com.azure.storage.common.credentials.SharedKeyCredential;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Locale;
+
+/**
+ * This example shows how to start using the Azure Storage Blob SDK for Java.
+ */
+public class BasicExample {
+
+    public static void main(String[] args) {
+
+        // From the Azure portal, get your Storage account's name and account key.
+        String accountName = getAccountName();
+        String accountKey = getAccountKey();
+
+        // Use your Storage account's name and key to create a credential object; this is used to access your account.
+        SharedKeyCredential credential = new SharedKeyCredential(accountName, accountKey);
+
+        /**
+         * From the Azure portal, get your Storage account blob service URL endpoint.
+         * The URL typically looks like this:
+         */
+        String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
+
+        // Create a StorageClient object that wraps the service endpoint, credential and a request pipeline.
+        StorageClient storageClient = StorageClient.storageClientBuilder().endpoint(endpoint).credential(credential).buildClient();
+
+        // This example shows several common operations just to get you started.
+
+        /**
+         * Create a client that references a to-be-created container in your Azure Storage account. This returns a
+         * ContainerClient object that wraps the container's endpoint, credential and a request pipeline (inherited from storageClient).
+         * Note that container names require lowercase.
+         */
+        ContainerClient containerClient = storageClient.getContainerClient("myjavacontainerbasic" + System.currentTimeMillis());
+
+        // Create a container in Storage blob account.
+        containerClient.create();
+
+        /**
+         * Create a client that references a to-be-created blob in your Azure Storage account's container.
+         * This returns a BlockBlobClient object that wraps the blob's endpoint, credential and a request pipeline
+         * (inherited from containerClient). Note that blob names can be mixed case.
+         */
+        BlockBlobClient blobClient = containerClient.getBlockBlobClient("HelloWorld.txt");
+
+        String data = "Hello world!";
+        InputStream dataStream = new ByteArrayInputStream(data.getBytes());
+
+        /**
+         * Create the blob with string (plain text) content.
+         * NOTE: The stream containing the data must be replayable to support retries. That is, it must
+         * yield the same data every time it is subscribed to.
+         * NOTE: If the provided length does not match the actual length, this method will throw.
```

Comment:
I think you can remove the two notes here. The first one because we need to figure out retry support on these InputStreams since they aren't seekable/markable by default. The second one because this behavior may not be the case for the first preview (though you should confirm with Anu whether that change made it into Core)

Reply:
Removed the first one and the second notes will be removed too for now. An issue open to validate the length after preview 1.
https://github.com/Azure/azure-sdk-for-java/issues/4193

Will add the notes back once it is done.